addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -165,6 +176,58 @@ export class PosStore extends Reactive {
         return !this.cashier ? "LoginScreen" : "ProductScreen";
     }
 
+    setActivityListeners() {
+        IDLE_TIMER_SETTER = this.setIdleTimer.bind(this);
+        for (const event of NON_IDLE_EVENTS) {
+            window.addEventListener(event, IDLE_TIMER_SETTER);
+        }
+    }
+
+    shouldResetIdleTimer() {
+        const stayPaymentScreen =
+            this.mainScreen.component === PaymentScreen && this.get_order().payment_ids.length > 0;
+        return !stayPaymentScreen;
+    }
+
+    setIdleTimer() {
+        clearTimeout(this.idleTimer);
+        const isFloorOrScreenSaver = ["FloorScreen", "ScreenSaver", "LoginScreen"].includes(
+            this.mainScreen.component.name
+        );
+        let targetScreen = "ScreenSaver";
+        let timeoutDuration = 300000;
+        if (this.config.module_pos_restaurant && !isFloorOrScreenSaver) {
+            targetScreen = "FloorScreen";
+            timeoutDuration = 180000;
+        } else if (this.mainScreen.component.name == "LoginScreen") {
+            timeoutDuration = 120000;
+        }
+        if (this.shouldResetIdleTimer()) {
+            this.idleTimer = setTimeout(() => this.actionAfterIdle(targetScreen), timeoutDuration);
+        }
+    }
```

Comment:
```suggestion
    setIdleTimer() {
        clearTimeout(this.idleTimer);
        const isRestaurant = this.config.module_pos_restaurant;
        const screenName = this.mainScreen.component.name;
        const order = this.get_order();
        const isIdle = ["FloorScreen", "ScreenSaver", "LoginScreen"].includes(screenName);

        let target = "ScreenSaver";
        let duration = 300000;

        if (isRestaurant && !isIdle) {
            target = "FloorScreen";
            duration = 180000;
        } else if (this.mainScreen.component.name == "LoginScreen") {
            duration = 120000;
        }

        if (screenName === "PaymentScreen" && order.payment_ids.length > 0) {
            this.idleTimer = setTimeout(() => this.actionAfterIdle(target), duration);
        }
    }
```

You can also remove the override of `shouldResetIdleTimer` in `pos_hr_restaurant`since its managed here now.